ecpy_pulses/manifest.enaml

```diff
@@ -0,0 +1,678 @@
+
+    path : unicode, optional
+        Path to the file holding the sequence config. The stored sequence
+        should be a RootSequence not a TemplateSequence. Used in file mode.
+
+    prefs : dict, optional
+        Sequence config describing the sequence to rebuild. The stored sequence
+        should be a RootSequence not a TemplateSequence. Used in file mode.
+
+    Returns
+    -------
+    sequence : BaseSequence
+
+    """
+    manager = event.workbench.get_plugin('ecpy.pulses')
+    mode = event.parameters.get('mode')
+    if mode == 'ui':
+        parent_ui = event.parameters.get('parent_ui')
+        root = event.parameters['root']
+        dial = BuilderView(parent_ui, manager=manager)
+        dial.exec_()
+        if dial.result:
+            seq = dial.config.build_sequence()
+            if seq:
+                return seq
+            else:
+                BasicErrorsDisplay(errors=dial.config.errors,
+                                   kind='Config errors').exec_()
+
+    elif mode == 'file':
+        if 'path' in event.parameters:
+            prefs = ConfigObj(event.parameters['path'])
+        elif 'prefs' in event.parameters:
+            prefs = event.parameters['prefs']
+        else:
+            raise ValueError('No valid kwargs found for file mode')
+
+        # Collect the dependencies
+        core = event.workbench.get_plugin('enaml.workbench.core')
+        cmd = 'ecpy.app.dependencies.analyse'
+
+        cont = core.invoke_command(cmd, {'obj': prefs}) 
+        if cont.errors:
+            raise RuntimeError('Failed to analyse dependencies :\n%s' %
+                               cont.errors)
+
+        cmd = 'ecpy.app.dependencies.collect'
+        cont = core.invoke_command(cmd, {'kind': 'build',
+                                         'dependencies': cont.dependencies})
+        if cont.errors:
+            raise RuntimeError('Failed to collect dependencies :\n%s' %
+                               cont.errors)
+        build_dep = cont.dependencies
+
+        # Implement the logic here to rebuild the sequence
+        seq = RootSequence.build_from_config(deepcopy(prefs), build_dep)
+        return seq
+
+
+def build_context(event):
+    """ Select a context for a given pulse sequence.
+
+    """
+    manager = event.workbench.get_plugin('ecpy.pulses')
+    parent_ui = event.parameters.get('parent_ui')
+    root = event.parameters['root']
+    dial = ContextBuilderView(parent_ui, manager=manager)
+    dial.exec_()
+    if dial.result:
+        root.context = dial.context()
+
+# =============================================================================
+# --- Descriptions ------------------------------------------------------------
+# =============================================================================
+
+SHOW_MANAGER_DESCRIPTION = \
+""" Show the task manager window."""
+
+SHOW_ERRORS_DESCRIPTION = \
+""" Show the errors which happened on plugin startup."""
+
+CONTEXTS_REQUEST_DESCRIPTION = \
```

Comment:
Provide a full docstring (with parameters) (you can copy paste from the plugin).